File: easy/1784_check_if_binary_string_has_at_most_one_segment_of_ones/solution.py
class Solution:
    def checkOnesSegment(self, s: str) -> bool:
        has_ones_segment = False

        i = 0
        while i < len(s):
            if s[i] == "1":
                if has_ones_segment:
                    return False

                has_ones_segment = True

                while i + 1 < len(s) and s[i + 1] == "1":
                    i += 1

            i += 1

        return has_ones_segment

File: easy/1784_check_if_binary_string_has_at_most_one_segment_of_ones/solution.cpp
#include <string>

class Solution {
public:
  bool checkOnesSegment(std::string s) {
    const size_t sLen = s.size();

    bool hasOnesSegment = false;

    for (size_t i = 0; i < sLen; ++i) {
      if (s[i] == '1') {
        if (hasOnesSegment)
          return false;
        hasOnesSegment = true;

        while (i + 1 < sLen && s[i + 1] == '1')
          ++i;
      }
    }

    return hasOnesSegment;
  }
};
